### PR TITLE
[incubator-kie-issues-1131] test migration from V7 to code generation-20

### PR DESCRIPTION
**Pull request by Abhitocode**
In addition to previous changes associated,
Migrated tests within FlowTest.java

The test can be identified by referring to FlowTest.java:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java

Closes https://github.com/apache/incubator-kie-issues/issues/1131.

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#3` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3627 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/3/display/redirect

**Test results:**
- PASSED: 3355
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/3/testReport/org.jbpm.bpmn2/FlowTest/testMultipleGatewaysProcess/">org.jbpm.bpmn2.FlowTest.testMultipleGatewaysProcess</a></summary>
<br/>expected: 2<br/> but was: 1
</details>

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3627 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/2/display/redirect

**Test results:**
- PASSED: 3353
- FAILED: 3

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/2/testReport/org.jbpm.bpmn2/FlowTest/testMultipleGatewaysProcess/">org.jbpm.bpmn2.FlowTest.testMultipleGatewaysProcess</a></summary>
<br/>expected: 2<br/> but was: 1
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/2/testReport/org.jbpm.bpmn2/IntermediateEventTest/testIntermediateCatchEventTimerCycle3/">org.jbpm.bpmn2.IntermediateEventTest.testIntermediateCatchEventTimerCycle3</a></summary>
<br/>expected: 3<br/> but was: 2
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/2/testReport/org.kie.kogito.process.dynamic/DynamicCallResourceTest/testConcurrentDynamicCalls/">org.kie.kogito.process.dynamic.DynamicCallResourceTest.testConcurrentDynamicCalls</a></summary>
java.lang.AssertionError: 1 expectation failed.<br/>Expected status code <200> but was <400>.<br/>
</details>

**Review by elguardian — COMMENTED** [review]
LGTM

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java [hunk 1]
```diff
@@ -1435,15 +1377,19 @@ public void testExclusiveSplitDefaultNoCondition() throws Exception {
...
+        model.setAction("CreateAgent");
+
+        ProcessTestHelper.registerProcessEventListener(app, new DefaultKogitoProcessEventListener() {
+            ProcessInstance<MultipleGatewaysProcessModel> pi;
 
             @Override
             public void afterNodeTriggered(ProcessNodeTriggeredEvent event) {
                 if (event.getNodeInstance().getNodeName().equals("CreateAgent")) {
-                    pi.signalEvent("Signal_1", null);
+                    pi.send(Sig.of("Signal_1", null));
```
Comment (elguardian) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java:1392]: use the kruntime in here. not the pi.

**Comment by elguardian** [comment]
plz be aware of the CI failing

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java (hunk 1 above)
Comment (Abhitocode) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java:1392]: Okay

**Comment by kie-ci3** [comment]
**PR job** `#4` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3627 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/4/display/redirect

**Test results:**
- PASSED: 3355
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3627/4/testReport/org.jbpm.bpmn2/IntermediateEventTest/testEventBasedSplit2/">org.jbpm.bpmn2.IntermediateEventTest.testEventBasedSplit2</a></summary>
<br/>Expected size: 2 but was: 1 in:<br/>[EventDesciption [event=timerTriggered, nodeId=_4, nodeName=timer, eventType=timer, nodeInstanceId=e9261cd5-dcef-4689-9553-feae88c1cbe9, processInstanceId=dc3f100b-5879-4f3d-80d2-854ee09f6cb5, dataType=null, properties={TimerID=c5c33f3b-2355-4c8f-879a-f59c8c7550c8, Period=null, Delay=500ms, Date=null}]]
</details>

**Review by elguardian — APPROVED** [review]
LGTM

**Review by martinweiler — APPROVED** [review]
Great job @Abhitocode

**Pull request merged**